### ISSUE TITLE
Sky grid fixes

### ISSUE THREAD

**Issue by pannarale** [gwastro/pycbc#4610]
Regarding https://github.com/gwastro/pycbc/pull/4485 , let's
- [ ] rely on already existing functions for angular/cartesian conversions, if possible
- [x] check conventions (coordinates especially)
- [x] fix a small boundary effect issue with the current circular grid
- [x] improve beyond ~~circular~~ uniform grid, ideally with the ideas in https://www.arxiv.org/abs/1208.2340 and https://arxiv.org/abs/1307.4158 or with the approximate (but less approximate than what is currently implemented) https://arxiv.org/abs/1509.03426
- [x] PEP8 style. This can largely be done automatically, see https://github.com/gwastro/pycbc/wiki/How-to-satisfy-CodeClimate.

**Comment by titodalcanton** [comment]
PEP8 should have (mostly) been addressed in #4988.

**Comment by pannarale** [comment]
- [ ] At line 90, `sky_grid.py`  does `detectors = hf.attrs['detectors']`, but `pycbc_make_sky_grid` fills ins `hf['instruments'] = [d for d in args.instruments]` at line 187.  So if you generate a sky grid with `pycbc_make_sky_grid` and pass it to `pycbc_multi_inspiral`, it will fail because `sky_grid.py` expects `'detectors'`.  I leave it to @titodalcanton to decide which of the two ends to modify.

**Comment by titodalcanton** [comment]
@pannarale indeed, thanks for catching that. And a similar problem exists with the `ref_gps_time` / `trigger_time` attributes.

**Comment by titodalcanton** [comment]
#5000 also ticks the second checkbox.

**Comment by pannarale** [comment]
For reference, [here](https://ldas-jobs.ligo.caltech.edu/~francesco.pannarale/LVC/pygrb_dec2024_1/) is a run on multiple sky points.
